lib/watchlist_edits.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/wiki_edits"
+class WatchlistEdits < WikiEdits
+  def initialize(wiki = nil, my_array = nil)
+    super(wiki)
+    @my_array = my_array || []
+  end
+
+  def oauth_credentials_valid?(current_user)
```

Comment:
A method like this, by convention, should answer the question it asks. I would expect this simply to return `true` or `false` without any actions that aren't necessary to answer that question. The method name does not give any hint that it actually also executes adding to the watchlist.